src/OrchardCore/OrchardCore.ContentManagement.GraphQL/ServiceCollectionExtensions.cs

```diff
@@ -32,6 +33,15 @@ public static IServiceCollection AddContentGraphQL(this IServiceCollection servi
             return services;
         }
 
+        public static IServiceCollection AddContentFieldsInputGraphQL(this IServiceCollection services)
+        {
+            services.AddScoped<IIndexAliasProvider, DynamicContentFieldsIndexAliasProvider>();
+            services.AddScoped<IContentTypeBuilder, DynamicContentTypeWhereInputBuilder>();
+            services.AddScoped<IContentDefinitionEventHandler, DynamicContentFieldsIndexAliasProvider>();
```

Comment:
Be careful if some of those new types are being resolved while initializing the schema. In that case they must be transient or singleton.

Reply:
It is recommended that these notes be written to the code. After all, OC's current code is huge, and we have little documentation on the logic and notes in the code

Graphql is a very complex module in OC

I've been looking at the code for enabling functionality from recipes to analyze the cause of this issue #16214, and it's just too complicated!

Reply:
> It is recommended that these notes be written to the code. After all, OC's current code is huge, and we have almost no comments on the logic code and related documentation

@Piedone , What do you think?

Reply:
Generally I agree, but what are you asking about, specifically?

If there's anything to keep in mind when registering GraphQL-related services, I'd rather make it explicit, similar to what we do with e.g. `AddContentPart()` and `AddDataMigration()`